agent/src/softwarecontainerfactory.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "softwarecontainerfactory.h"
+
+namespace softwarecontainer {
+std::shared_ptr<SoftwareContainerAbstractInterface>
+SoftwareContainerFactory::createContainer(const ContainerID id,
+                                          std::unique_ptr<const SoftwareContainerConfig> config)
+{
+    auto container = std::shared_ptr<SoftwareContainerAbstractInterface>
+                                    (new SoftwareContainer(id, std::move(config)));
```

Comment:
I think it would be more similar to other parts of the code if the opening parenthesis is on the line above. It's basically the same as when you have a method and need to break the line at the first arg:

longMethodName(
    arg1, arg2);